### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by OR13**
- allow multiple proof types
- address editors notes

### CODE REVIEW

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 1]
```diff
@@ -160,21 +160,20 @@ while switching from one verifiable data structure to another, so long as both s
...
 with the following initial contents:
 
-| Identifier  | Proof Type   | Reference
+| Identifier  | Proof Type   | Proof Value | Reference
 |---
-|0            | N/A          |
-|TBD_2        | inclusion    | {{sec-generic-inclusion-proof}}
-|TBD_3        | consistency  | {{sec-generic-consistency-proof}}
+|0            | N/A          | N/A | N/A
+|TBD_2        | inclusion    | bstr or list of bstr | {{sec-generic-inclusion-proof}}
```
Comment (JAG-UK) [inline draft-ietf-cose-merkle-tree-proofs.md:166]: Wouldn't it be better/simpler for this to ALWAYS be a list and sometimes only have 1 element?

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 2]
```diff
@@ -160,21 +160,20 @@ while switching from one verifiable data structure to another, so long as both s
...
 
-Editors note: The registry requirements needs to address the case of multiple proofs of a given type.
-
 ## Inclusion Proof {#sec-generic-inclusion-proof}
 
 Inclusion proofs provide a mechanism for a verifier to validate set membership.
 
 The integer identifier for this Proof Type is TBD_2.
 The string identifier for this Proof Type is "inclusion".
+The value of this Proof Type is bstr or list of bstr.
```
Comment (JAG-UK) [inline draft-ietf-cose-merkle-tree-proofs.md:176]: Wouldn't it be better/simpler for this to ALWAYS be a list and sometimes only have 1 element?

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 3]
```diff
@@ -184,6 +183,7 @@ Consistency proofs provide a mechanism for a verifier to validate the consistenc
 
 The integer identifier for this Proof Type is TBD_3.
 The string identifier for this Proof Type is "consistency".
+The value of this Proof Type is bstr or list of bstr.
```
Comment (JAG-UK) [inline draft-ietf-cose-merkle-tree-proofs.md:186]: Wouldn't it be better/simpler for this to ALWAYS be a list and sometimes only have 1 element?

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 4]
```diff
@@ -195,8 +195,8 @@ are mapped to the terminology defined in this document, using cbor and cose.
 RFC9162_SHA256 requires the following:
 
 - TBD_1 (verifiable-data-structure): 1, the integer representing the RFC9162_SHA256 verifiable data structure algorithm.
-- TBD_2 (inclusion-proof): a bstr representing the RFC9162_SHA256 inclusion proof
-- TBD_3 (consistency-proof): a bstr representing the RFC9162_SHA256 consistency proof
+- TBD_2 (inclusion-proof): a bstr or list of bstr representing RFC9162_SHA256 inclusion proofs
```
Comment (JAG-UK) [inline draft-ietf-cose-merkle-tree-proofs.md:198]: Wouldn't it be better/simpler for this to ALWAYS be a list and sometimes only have 1 element?

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 5]
```diff
@@ -195,8 +195,8 @@ are mapped to the terminology defined in this document, using cbor and cose.
...
+- TBD_3 (consistency-proof): a bstr or list of bstr representing RFC9162_SHA256 consistency proofs
```
Comment (JAG-UK) [inline draft-ietf-cose-merkle-tree-proofs.md:199]: Wouldn't it be better/simpler for this to ALWAYS be a list and sometimes only have 1 element?

**Review by JAG-UK — COMMENTED** [review]
I have made the same single comment 5 times. Sorry about that, but it has many tentacles.

File: draft-ietf-cose-merkle-tree-proofs.md (hunk 1 above)
Comment (OR13) [reply draft-ietf-cose-merkle-tree-proofs.md:166]: I agree, this was just my first attempt at an update... and I wanted to hear from @henkbirkholz and other CBOR / COSE experts on how best to address this.

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 6]
```diff
@@ -163,8 +163,8 @@ with the following initial contents:
 | Identifier  | Proof Type   | Proof Value | Reference
 |---
 |0            | N/A          | N/A | N/A
-|TBD_2        | inclusion    | bstr or list of bstr | {{sec-generic-inclusion-proof}}
```
Comment (henkbirkholz) [inline draft-ietf-cose-merkle-tree-proofs.md]: Technically, that would be:

```
bstr / [ 2* bstr ]
```
or
```
one-or-more<T> = T / [ 2* T ]
one-or-more<bstr> 
```
and not
```
[ + bstr ]
```

as illustrated in https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/issues/109#issue-1935673877

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 7]
```diff
@@ -320,6 +326,16 @@ The latest Merkle tree hash as defined in {{-certificate-transparency-v2}}.
 
 The payload MUST be attached.
 
+~~~~ cddl
+
+consistency-proofs = [ + bstr ]
+
+unprotected-header-map = {
+  &(consistency-proof: TBD_2) => consistency-proofs
```
Comment (OR13) [inline draft-ietf-cose-merkle-tree-proofs.md:334]:
```suggestion
  &(consistency-proof: TBD_3) => consistency-proofs
```

File: draft-ietf-cose-merkle-tree-proofs.md [hunk 8]
```diff
@@ -306,10 +313,9 @@ The protected header for an RFC9162_SHA256 consistency proof signature is:
 
 * alg (label: 1): REQUIRED. Signature algorithm identifier. Value type: int / tstr.
 * verifiable-data-structure (label: TBD_1): REQUIRED. verifiable data structure algorithm identifier. Value type: int / tstr.
+* kid (label: 4): OPTIONAL. Key identifier. Value type: bstr
 * crit (label: 2): OPTIONAL. Criticality marker. Value type: [ +label ]
```
Comment (henkbirkholz) [inline draft-ietf-cose-merkle-tree-proofs.md:317]:
```suggestion
* kid (label: 4): OPTIONAL. Key identifier. Value type: bstr
* crit (label: 2): OPTIONAL. Criticality marker. Value type: [ + label ]
```

**Pull request merged**